src/tron/tronPlugin.ts

```diff
@@ -44,7 +73,13 @@ export class TronPlugin extends CurrencyPlugin {
   }
 
   async derivePublicKey(walletInfo: EdgeWalletInfo): Promise<Object> {
-    throw new Error('Must implement checkBlockchainInnerLoop')
+    const type = walletInfo.type.replace('wallet:', '')
+    if (type === this.currencyInfo.pluginId) {
```

Comment:
This is wrong. You should be doing `if (walletType !== this.currencyInfo.walletType) { throw ... }`, instead of slicing up the strings.